src/time_compat.erl

```diff
@@ -43,160 +43,242 @@
 %% where it has not yet been deprecated.
 %%
 
+%% Declare versioned functions to allow dynamic code loading,
+%% depending on the Erlang version running. See 'code_version.erl' for details
+-version_support(
+   [{monotonic_time, monotonic_time_pre_18, monotonic_time_post_18, 0},
+    {monotonic_time, monotonic_time_pre_18, monotonic_time_post_18, 1},
+    {erlang_system_time, erlang_system_time_pre_18, erlang_system_time_post_18, 0},
+    {erlang_system_time, erlang_system_time_pre_18, erlang_system_time_post_18, 1},
+    {os_system_time, os_system_time_pre_18, os_system_time_post_18, 0},
+    {os_system_time, os_system_time_pre_18, os_system_time_post_18, 1},
+    {time_offset, time_offset_pre_18, time_offset_post_18, 0},
+    {time_offset, time_offset_pre_18, time_offset_post_18, 1},
+    {convert_time_unit, convert_time_unit_pre_18, convert_time_unit_post_18, 0},
```

Comment:
The infinite loop comes from this line: `convert_time_unit` arity is 3, not 0. Thus, convert_time_unit is not replaced.